flocker/ca/functional/test_validation.py

```diff
@@ -49,14 +49,12 @@ def __init__(self):
 
     def dataReceived(self, data):
         self._buffer += data
-        if self._buffer == EXPECTED_STRING:
-            self.result.callback("handshake!")
-            self.result = None
 
     def connectionLost(self, reason):
-        if self.result is not None:
+        if self._buffer == EXPECTED_STRING:
+            self.result.callback("handshake!")
+        else:
             self.result.errback(reason)
```

Comment:
Probably would be useful to say something about `self._buffer` as well here (though I expect the typical reason this test would fail is TLS handshake failure ... which is probably mostly represented by `reason`).

:question: